### ISSUE TITLE
`vue/script-setup-uses-vars` is not working when generating with vue cli

### ISSUE THREAD

**Issue by Cpaczek** [vuejs/eslint-plugin-vue#1785]
**Checklist**

- [x ] I have tried restarting my IDE and the issue persists.
- [ x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version: ^7.32.0** 
- **eslint-plugin-vue version: 8.4.0** 
- **Node version:v14.18.1** 
- **Operating System:Windows 10** 

**Please show your full configuration:**

```json5
module.exports = {
  root: true,
  env: {
    node: true,
  },
  parserOptions: {
    ecmaVersion: 2021,
    parser: "@typescript-eslint/parser",
    sourceType: "module",
  },
  parser: "vue-eslint-parser",
  extends: [
    "eslint:recommended",
    "@vue/typescript/recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:vue/vue3-recommended",
    "@vue/prettier",
    "@vue/prettier/@typescript-eslint",
  ],
  rules: {
    "no-console": process.env.NODE_ENV === "production" ? "warn" : "off",
    "no-debugger": process.env.NODE_ENV === "production" ? "warn" : "off",
    "vue/script-setup-uses-vars": "error"
  },
};

```

**What did you do?**	

```vue
<script setup lang="ts">
import { reactive } from "vue";
const data = reactive({
  type: 1,
  title: "Hello BalmUI",
  openDrawer: false,
});
</script>
```

**What did you expect to happen?**
Ignore the unused vaiable because its in <script setup>

**What actually happened?**
```
warning: 'data' is assigned a value but never used (@typescript-eslint/no-unused-vars) at src\App.vue:37:7:
  35 | <script setup lang="ts">
  36 | import { reactive } from "vue";
> 37 | const data = reactive({
     |       ^
  38 |   type: 1,
  39 |   title: "Hello BalmUI",
  40 |   openDrawer: false,
```

**Repository to reproduce this issue**
https://github.com/Cpaczek/eslint-example-repo

**Comment by FloEdelmann** [comment]
That has nothing to do with Vue CLI. The lint error is correct. You define the `data` variable, and never use it in the template or anywhere else.

---

Maybe you wanted to do this instead?

```vue
<script setup lang="ts">
import { ref } from "vue";
const type = ref(1);
const title = ref("Hello BalmUI");
const openDrawer = ref(false);
</script>
```

But still you'd have to use `type`, `title` and `openDrawer` somewhere in the template or anywhere else then to not trigger this lint error.
